webapp/src/components/resultPanel/index.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {forwardRef} from 'react';
+
+import SVGWrapper from 'components/svgWrapper';
+import SVGIcons from 'plugin_constants/icons';
+
+type ResultPanelProps = {
+    iconClass?: string | null;
+    header?: string | null;
+    className?: string;
+    primaryBtnText?: string;
+    secondaryBtnText?: string;
+    onPrimaryBtnClick?: () => void | null;
+    onSecondaryBtnClick?: () => void | null;
+};
+
+const ResultPanel = forwardRef<HTMLDivElement, ResultPanelProps>(({
+    header,
+    className = '',
+    primaryBtnText,
+    secondaryBtnText,
+    onPrimaryBtnClick,
+    onSecondaryBtnClick,
+    iconClass,
+}: ResultPanelProps, feedAddedRef): JSX.Element => (
+    <div
+        className={`modal__body modal-body d-flex align-items-center justify-content-center flex-column secondary-panel ${className}`}
+        ref={feedAddedRef}
+    >
+        <>
+            {iconClass ? (
+                <i className={`fa result-panel-icon ${iconClass ?? 'fa-check-circle-o'}`}/>
+            ) : (
+                <SVGWrapper
+                    className='result-panel-icon'
+                    width={58}
+                    height={58}
+                    viewBox='0 0 58 58'
+                >
+                    {SVGIcons.check}
+                </SVGWrapper>
+            )}
+            <h3 className='result-panel-text'>{header ?? 'Add new'}</h3>
+            {onPrimaryBtnClick && (
+                <button
+                    className='btn btn-primary'
+                    onClick={onPrimaryBtnClick}
+                >
+                    {primaryBtnText ?? 'Create new'}
```

Comment:
Is the value of 'primaryBtnText' ever a empty string?

Reply:
it is an optional prop, so it can be undefined

Reply:
Can be undefined but not empty right?

Reply:
updated